modules/users/server/controllers/user-crud.controller.js

```diff
@@ -269,44 +270,70 @@ function userCrudController(logger, shared) {
    * @param {Response} res  The Express response object
    */
   function adminUpdate(req, res) {
+    if(req.user.role != roleConfig.ADMIN_ROLE_NAME){
+      res.status(403).send();
+      return;
+    }   
+
     if (!isAuthorized(req.user)) {
       res.status(403).send({ success: false, message: "Forbidden" });
       return;
     }
 
-    let deferred = q.defer();
     let user = req.body;
-
+    //update role if necessary
     return new Promise((resolve, reject) => {
-      //note that this doesn't affect user roles
-      let updateDef = {
-        $set: {
-          firstName: user.firstName,
-          lastName: user.lastName,
-          updated: new Date()
+      if (user.role) {
+        self.roleModule.determineSubroles(user.role)
+          .then(subroles => {
+            user.subroles = subroles;
+            //resolve after subroles are determined
+            resolve();
+          })
+          .catch(error => {
+            reject(error);
```

Comment:
catching an error just to reject it isn't really needed. You can just omit the catch and it would have the same behavior